Kinvey/Kinvey/CustomEndpoint.swift

```diff
@@ -36,13 +46,77 @@ open class CustomEndpoint: NSObject {
         return request
     }
     
+    //MARK: Mappable
+    
+    /// Executes a custom endpoint by name and passing the expected parameters.
+    @discardableResult
+    internal static func execute<T: Mappable>(_ name: String, params: JsonDictionary? = nil, client: Client = sharedClient, completionHandler: CompletionTHandler<T>? = nil) -> Request {
+        let request = callEndpoint(name, params: params, client: client) { data, response, error in
+            if let completionHandler = dispatchAsyncMainQueue(completionHandler) {
+                if let response = response , response.isOK, let obj: T = client.responseParser.parse(data) {
+                    completionHandler(obj, nil)
+                } else {
+                    completionHandler(nil, buildError(data, response, error, client))
+                }
+            }
+        }
+        return request
+    }
+    
+    /// Executes a custom endpoint by name and passing the expected parameters.
+    @discardableResult
+    internal static func execute<T: Mappable>(_ name: String, params: JsonDictionary? = nil, client: Client = sharedClient, completionHandler: ArrayCompletionTHandler<T>? = nil) -> Request {
+        let request = callEndpoint(name, params: params, client: client) { data, response, error in
+            if let completionHandler = dispatchAsyncMainQueue(completionHandler) {
+                if let response = response , response.isOK, let objArray: [T] = client.responseParser.parse(data) {
+                    completionHandler(objArray, nil)
+                } else {
+                    completionHandler(nil, buildError(data, response, error, client))
+                }
+            }
+        }
+        return request
+    }
+    
+    //MARK: StaticMappable
+    
+    /// Executes a custom endpoint by name and passing the expected parameters.
+    @discardableResult
+    internal static func execute<T: StaticMappable>(_ name: String, params: JsonDictionary? = nil, client: Client = sharedClient, completionHandler: CompletionTHandler<T>? = nil) -> Request {
```

Comment:
ObjectMapper documentation warns to don't extend from `BaseMappable`, so I basically would like to avoid users making this mistake forcing then to always extend from `Mappable` or `StaticMappable`